cmd/builder/internal/command.go

```diff
@@ -100,11 +110,44 @@ func initConfig() error {
 		cfg.Logger.Error("failed to load env var", zap.Error(err))
 	}
 
-	if err := k.UnmarshalWithConf("", &cfg, koanf.UnmarshalConf{Tag: "mapstructure"}); err != nil {
+	cfgFromFile := builder.Config{}
+	if err := k.UnmarshalWithConf("", &cfgFromFile, koanf.UnmarshalConf{Tag: "mapstructure"}); err != nil {
 		cfg.Logger.Error("failed to unmarshal config", zap.Error(err))
 		return err
 	}
 
+	cfg.Exporters = cfgFromFile.Exporters
+	cfg.Extensions = cfgFromFile.Extensions
+	cfg.Receivers = cfgFromFile.Receivers
+	cfg.Processors = cfgFromFile.Processors
+	cfg.Replaces = cfgFromFile.Replaces
+	cfg.Excludes = cfgFromFile.Excludes
+
+	if !flags.Changed(SkipComplicationFlag) && cfgFromFile.SkipCompilation {
```

Comment:
There is some cleanup to do here but wanted to get the basic idea out there:

1. Let cobra set the flags in cfg struct. 
2. Read in the config file
3. Apply all entries to cfg from the config file that cannot be overridden by a flag
4. Check if a flag has not been changed and if the config file included a value. If true, use the config file value. 

We can't read the config file right into cfg or we'll overwrite potentially set flag values.